code/go/0chain.net/smartcontract/storagesc/models.go

```diff
@@ -342,6 +342,7 @@ type StorageNode struct {
 	SavedData       int64                  `json:"saved_data"`
 	// StakePoolSettings used initially to create and setup stake pool.
 	StakePoolSettings stakepool.StakePoolSettings `json:"stake_pool_settings"`
+	TotalStake        int64                       `json:"total_stake"`
```

Comment:
Run `make msgp` to update the `_gen.go` file, if you don't have msgp installed, please run `make install-msgp` in advance. 

I don't see the `TotalStake` is being used other than the event db, please update related code that would update it, otherwise, we'd better keep the MPT data struct untouched.

Reply:
@peterlimg maybe you missed https://github.com/0chain/0chain/pull/1078/files#diff-ce2ed4fb820ddf8bc8517075d5242e73b2e7547262aababb0be2e39ea481a56dR463

But thanks for the msgp tip will do that

Reply:
also the total stake is updated https://github.com/0chain/0chain/pull/1078/files#diff-ce2ed4fb820ddf8bc8517075d5242e73b2e7547262aababb0be2e39ea481a56dR469
Here if that is what you missed... not sure if thatis what you were searching for

Reply:
Hi @samtholiya, can you share the newAllocation PR. BTW, the code you referenced is still the place for update the total stake in event db. 

I suppose there are some place (perhaps block rewards) the `blobbler.TotalStake` is updated and saved to MPT, then emit the total stake changes db events, then save it to events db, but in this PR I  only see the `TotalStake` field is added to MPT struct, and only event db is using it.  @Sriep can you check

Reply:
We don't need to keep ToptalStake in `StorageNode`, as we can always use [stakePool.stakes()](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/storagesc/stakepool.go#L101) instead.  Whichever, the important thing is that we emit a `TagUpdateBlobber` event every time a delegate pool's balance changes.

We will at least need to know the total stake in the new allocation request sc in order to validate the input blobbers. We do not need it for paying rewards, as in block rewards as this is added to a separate reward account, not the pool's balance.

Reply:
If the `TotalStake` is only used by `GetBlobberHandler` and for API only, we'd better it out of MPT, the principle is if not necessary, please do not change MPT data struct.